scripts/linkchecker.py

```diff
@@ -410,10 +411,30 @@ def validate_links(page):
 
     matches = regex.findall(content)
     records = []
+    target_records = []
     for m in matches:
         r = check_target(page, m[0], m[1])
         if r:
             records.append(r)
+            target_records.append(m[1])
+
+    # if multiple records are the same they need not be checked repeatedly
+    target_records = set(target_records)
+
+    if in_place_edit and len(target_records) > 0:
+        updated_data = []
+        for line in data:
+            if any(rec in line for rec in target_records):
+                for rec in target_records:
+                    line = line.replace(
+                        "(" + rec + ")",
```

Comment:
```suggestion
                        f"({rec})",
```

Can we assume python 3.6 or later? (It was released in 2016.) If so, [f-Strings](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) can make this more readable.

Reply:
Not an expert in this script but I believe we cannot assume because it's going to be run locally. Although I imagine we could document that this requires Python 3.6+.